metrics/src/metrics.rs

```diff
@@ -281,23 +291,35 @@ impl MetricsAgent {
                     max_points,
                     max_points_per_sec,
                     last_write_time,
+                    receiver.len(),
                 );
                 last_write_time = now;
             }
         }
         trace!("run: exit");
     }
 
+    fn send(&self, command: MetricsCommand) {
+        match self.sender.try_send(command) {
+            Err(TrySendError::Full(cmd)) => {
+                warn!("metric queue full, current point: {:?}.", cmd);
+                // try again and block on sending
+                let _ = self.sender.send(cmd);
```

Comment:
On a bounded channel this will block if the channel is full.
We don't want to block validator processes just to submit metrics.
Please revert to `unbounded` channel since we do not intend to drop metrics anyways.

Also writing a `warn!` log with every data-point is extremely spammy.
If adding such `warn!` log was necessary, then it is at least better to be done in `write`, as opposed to here (so every 10 seconds, not with every datapoint).

But given that there is already a metric, I don't see much utility in this `warn` log; (there are tons of more critical things than the size of this channel to `warn!` log).

Reply:
Agree. 

If we make the queue not bounded, there is no points to do try_send/send. The only changes is to report queue length. 

To be clear, I created another pull question for just reporting. 
https://github.com/solana-labs/solana/pull/24661

There is another place that we dropped points in the metrics. Should we *not* truncate here too?
https://github.com/solana-labs/solana/blob/6ff16b17285b6870773433b806e5f1923eeafddb/metrics/src/metrics.rs#L205